packages/core/src/createContext.ts

```diff
@@ -165,11 +162,14 @@ export function updateContextByNormalizedConfig(
   context: RsbuildContext,
   config: NormalizedConfig,
 ) {
-  context.distPath = getAbsoluteDistPath(context.rootPath, config);
-
-  if (config.source.entry) {
-    context.entry = getEntryObject(config, 'web');
-  }
+  // Try to get the root distPath of context.environments as context.distPath
+  context.distPath = Object.values(context.environments).reduce((p, e) => {
+    // foo/dist/server vs foo/dist
+    if (!p || p.startsWith(e.distPath)) {
```

Comment:
Can this function get the common parent directory as expected?

For example, `/path/to/dist/client` and `/path/to/dist/server`, the parent directory is `/path/to/dist`

Reply:
but if `/path/dist-client` and `/path/dist-server`, the parent directory is `/path`, is expected?

Reply:
Yes, because we can't use `/path/dist-client' or `/path/dist-server'.